addons/crm_livechat/models/mail_channel.py

```diff
@@ -26,16 +26,30 @@ def _execute_command_lead(self, **kwargs):
         self._send_transient_message(partner, msg)
 
     def _convert_visitor_to_lead(self, partner, channel_partners, key):
+        """ Create a lead from channel /lead command
+        :param partner: internal user partner (operator) that created the lead;
+        :param channel_partners: channel members;
+        :param key: operator input in chat ('/lead Lead about Product')
+        """
         description = ''.join(
             '%s: %s\n' % (message.author_id.name or self.anonymous_name, message.body)
             for message in self.channel_message_ids.sorted('id')
         )
+        # filter out partners linked to a public user; easier than a search to avoid
+        # domain using user_ids.groups_id
+        customers = channel_partners.partner_id.filtered(lambda p:
+            p != partner and (any(not user._is_public() for user in p.user_ids) if p.user_ids else True)
+        )
+        # sort to have shared partners first so that other people in chat (operators or
+        # internal users) are not set as customers
+        customers = customers.sorted('partner_share', reverse=True)
```

Comment:
I don't know `partner_share` so I might be wrong, but don't we want to filter them out instead?

In a chat with guest + 2 operators/internal users, due to guest (public partner) being ignored, the other internal user will be considered as customer by being the first on the remaining list.

Reply:
This is similar to magic we have to do in JS actually, if the public partner is on the list, there should be no other partner considered as customer than the "guest".

Reply:
Actually first we filter out public (filtered hereabove), then we sort remaining people with partner_share first (partner_share = no user or portal user). That way portal users come first, public are out.

Reply:
About operator: not sure you can easily say "take partner in lead but remove operators". I could create a lead from a chat with a known user, why not. Or we simply filter out all internal users. Not sure what is best, especially for stable, and considering our standard use cases.

Reply:
I said on my second message, I think if public is in nobody else should be considered probably.
When public is inside, public is your customer. It's just that we don't want to "add it as customer", but we shouldn't look for anyone else.

Reply:
That's not what the code was doing in v14. If we consider public was filtered out before the patch adding public to channel members, lead command was trying to add other people (not the operator) as customer (crashing if more than one participant other than operator and public). Changing that could be considered as a functional change and may break existing workflows. Or we consider this is a corner case. Not sure what people do with livechat and we can be surprised of what people do with odoo side-features-or-bugs-or-not-sure ...

Reply:
I understand, I think it's not correct and my suggestion is better, but if you think it's safer to restore the previous non ideal situation than trying to make it "better" in a bug fix, that is perfectly reasonable as well.

So I'll leave it up to you for the final decision, as I don't CRM workflows or side-features-or-bugs-or-not-sure well enough to decide what is safe in this context.

Reply:
Well depends on how people use livechat; crm allows you to do whatever you want with your customer but afterwards leads may be merged if customer is the same ^^. If I summarize your proposal

 * public in it -> set nobody
 * public not in it -> set first share found

Works for me.